packages/services/test/src/kernel/ikernel.spec.ts

```diff
@@ -182,6 +182,96 @@ describe('Kernel.IKernel', () => {
 
   });
 
+  context('#anyMessage', () => {
+
+    it('should be emitted for an unhandled message', (done) => {
+      tester = new KernelTester();
+      tester.start().then(kernel => {
+        kernel.anyMessage.connect((k, args) => {
+          if (args.msg.header.msg_type === 'kernel_info_reply' ||
+              args.msg.header.msg_type === 'kernel_info_request') {
+            // Skip these leading messages
+            return;
+          }
+          expect(args.msg.header.msg_type).to.be('foo');
+          expect(args.direction).to.be('recv');
+          done();
+        });
+        let msg = KernelMessage.createShellMessage({
+          msgType: 'foo',
+          channel: 'shell',
+          session: kernel.clientId
+        });
+        msg.parent_header = msg.header;
+        tester.send(msg);
+      }).catch(done);
+    });
+
+    it('should be emitted for an iopub message', (done) => {
+      tester = new KernelTester();
+      tester.start().then(kernel => {
+        kernel.anyMessage.connect((k, args) => {
+          if (args.msg.header.msg_type === 'kernel_info_reply' ||
+              args.msg.header.msg_type === 'kernel_info_request') {
+            // Skip these leading messages
+            return;
+          }
+          expect(args.direction).to.be('recv');
+          done();
+        });
+        let msg = KernelMessage.createMessage({
+          msgType: 'status',
+          channel: 'iopub',
+          session: kernel.clientId
+        }) as KernelMessage.IStatusMsg;
+        msg.content.execution_state = 'idle';
+        msg.parent_header = msg.header;
+        tester.send(msg);
+      }).catch(done);
+    });
+
+    it('should be emitted for an stdin message', (done) => {
+      tester = new KernelTester();
+      tester.start().then(kernel => {
+        kernel.anyMessage.connect((k, args) => {
+          if (args.msg.header.msg_type === 'kernel_info_reply' ||
+              args.msg.header.msg_type === 'kernel_info_request') {
+            // Skip these leading messages
+            return;
+          }
+          expect(args.msg.content.value).to.be('foo');
+          expect(args.direction).to.be('send');
+          done();
+        });
+        kernel.sendInputReply({value: 'foo'});
+      }).catch(done);
+    });
+
+    it('should not be emitted for a different client session', () => {
```

Comment:
@vidartf - what is the purpose for this test? I think the test should be removed, and that anyMessage should emit on *any* message a kernel receives. That's what the code does.

(Note that I think when this PR was merged, these tests were not actually being run, so this test didn't show a failure.)

Reply:
(I'm removing this test in #4697 - I just wanted to make sure it made sense to remove it.)

Reply:
I agree this can be removed. It was a modified version of the similar one for `unhandledMessage` above, but I dont think that translates here.